db/system_keyspace.cc

```diff
@@ -1668,6 +1669,11 @@ future<> update_tokens(const std::unordered_set<dht::token>& tokens) {
     });
 }
 
+future<> update_cdc_streams_timestamp(db_clock::time_point tp) {
+    return execute_cql(format("INSERT INTO system.{} (key, cdc_streams_timestamp) VALUES (?, ?)", LOCAL), sstring(LOCAL), tp)
+            .discard_result().then([] { return force_blocking_flush(LOCAL); });
+}
```

Comment:
Why do we need a blocking flush here? It's not a disaster to have it, but better to avoid if not strictly needed.

Reply:
(because it slows down booting)

Reply:
I want to make sure the node is committed to using this particular generation timestamp before it finishes booting.
```
    // Persist the CDC streams timestamp before we persist bootstrap_state = COMPLETED.
    if (_cdc_streams_ts) {
        db::system_keyspace::update_cdc_streams_timestamp(*_cdc_streams_ts).get();
    }
    // If we crash now, we will choose a new CDC streams timestamp anyway (because we will also choose a new set of tokens).
    // But if we crash after setting bootstrap_state = COMPLETED, we will keep using the persisted CDC streams timestamp after restarting.

    db::system_keyspace::set_bootstrap_state(db::system_keyspace::bootstrap_state::COMPLETED).get();
    // At this point our local tokens and CDC streams timestamp are chosen (_bootstrap_tokens, _cdc_streams_ts) and will not be changed.
```

Reply:
> (because it slows down booting)

The slowdown is insignificant compared to the length of the rest of the boot process.